pkg/sbom/io/decode.go

```diff
@@ -363,28 +363,41 @@ func (m *Decoder) addLangPkgs(sbom *types.SBOM) {
 // addOrphanPkgs adds orphan packages.
 // Orphan packages are packages that are not related to any components.
 func (m *Decoder) addOrphanPkgs(ctx context.Context, sbom *types.SBOM) error {
-	osPkgMap := make(map[string]ftypes.Packages)
+	osPkgMap := make(map[ftypes.OS]ftypes.Packages)
 	langPkgMap := make(map[ftypes.LangType]ftypes.Packages)
 	for _, pkg := range m.pkgs {
 		p := (*purl.PackageURL)(pkg.Identifier.PURL)
 		switch p.Class() {
 		case types.ClassOSPkg:
-			osPkgMap[p.Type] = append(osPkgMap[p.Type], *pkg)
+			os := *p.OS()
+			osPkgMap[os] = append(osPkgMap[os], *pkg)
 		case types.ClassLangPkg:
 			langType := p.LangType()
 			langPkgMap[langType] = append(langPkgMap[langType], *pkg)
 		}
 	}
 
 	if len(osPkgMap) > 1 {
-		return xerrors.Errorf("multiple types of OS packages in SBOM are not supported (%q)", lo.Keys(osPkgMap))
+		oses := lo.MapToSlice(osPkgMap, func(os ftypes.OS, _ ftypes.Packages) string {
+			s := string(os.Family)
+			if os.Name != "" {
+				s = s + "-" + os.Name
+			}
+			return s
+		})
+		return xerrors.Errorf("multiple OSes in SBOM are not supported (%q)", oses)
```

Comment:
I thought about it.
We can leave this error.

But if we get problems about this - we will update the logic to get the OS (from `purl`) with the maximum number of packages.